theme/placeholder.scss

```diff
@@ -2,8 +2,15 @@
 // For licensing, see LICENSE.md or http://ckeditor.com/license
 
 .ck-placeholder::before {
+	position: absolute;
+	top: 0;
+	left: 0;
 	content: attr( data-placeholder );
 	cursor: text;
 	color: #c2c2c2;
 	pointer-events: none; // See ckeditor/ckeditor5#469.
 }
+
+.ck-placeholder {
```

Comment:
There is no perfect solution.

First, let me address `position: relative` doubt. This might break something only if the element previously had `position: absolute` or `position: fixed`. Even then, most of the times, the element will have a "stronger path" if it is styled by external CSS and that will overwrite our `position: relative`.

`position: absolute; left: 0; top: 0;` is more dangerous, I agree, however it works with `margin`s. Maybe we could try just setting `position: absolute` without `top` and `left`.